js/core.js

```diff
@@ -666,7 +666,46 @@ var ERMrest = (function (module) {
             }
 
             return result;
-        }
+        },
+
+        // figure out if Table is pure and binary association table.
+        // binary: Has 2 outbound foreign keys and there is a composite key constraint combining all the columns from both foreign keys.
+        // pure: There are no other columns apart from the foreign key columns and columns that are part of key and have serial4 type.
+        _isPureBinaryAssociation: function () {
+            if(this._isPureBinaryAssociation_cached === undefined) {
+                this._isPureBinaryAssociation_cached = this._computePureBinaryAssociation();
+            }
+            return this._isPureBinaryAssociation_cached;
+        },
+
+        _computePureBinaryAssociation: function () {
+            if (this.referredBy.length() > 0 || this.foreignKeys.length() != 2) {
+                return false; // not binary
+            }
+
+            var fkColset = new ColSet(this.foreignKeys.colsets().reduce(function(res, colset){
+                return res.concat(colset.columns);
+            }, [])); // set of foreignkey columns
+
+            var tempKeys = this.keys.all().filter(function(key) {
+                var keyCols = key.colset.columns;
+                return !(keyCols.length == 1 && keyCols[0].type.name == "serial4"  && !(keyCols[0] in fkColset.columns));
+            }); // the key that should contain foreign key columns.
+
+            if (tempKeys.length != 1 || !fkColset._equals(tempKeys[0].colset)) {
+                return false; // not binary
+            }
+
+            // check for purity:
+            for (var i = 0, col; i < this.columns.length(); i++) {
```

Comment:
These just check for columns that are involved in keys. But what if there is a column that is not part of any keys.
Based on what we defined, a table that has some columns that are not part of any keys, should not be considered as a pure binary association. 

If a table has a `modification_date` column that is not a foreign key column and is not part of any keys, should we still consider it as "pure binary association"?